test/unit/cms/s3_test.rb

```diff
@@ -65,6 +65,26 @@ class CMS::S3Test < ActiveSupport::TestCase
     assert_not CMS::S3.options[:use_fips_endpoint]
   end
 
+  test "#hostname returns nil when CMS::S3 is not enabled" do
+    CMS::S3.stubs(:enabled?).returns(false)
+
+    assert_nil CMS::S3.hostname
+  end
+
+  test "#hostname returns nil when it's not provided" do
+    CMS::S3.stubs(:config).returns(protocol: "https", hostname: "")
+
+    assert_nil CMS::S3.hostname
+  end
+
+  test "#hostname returns the correct config data when provided" do
+    hostname = "some_hostname"
+
+    CMS::S3.stubs(:config).returns(hostname: hostname)
```

Comment:
any reason to not just do `returns(hostname: 'some_hostname')` ?

Reply:
I think it is to reference the same string on line 85

Reply:
Ah, right, that's true, sorry :grimacing: